google/resource_google_project.go

```diff
@@ -198,30 +221,60 @@ func resourceGoogleProjectUpdate(d *schema.ResourceData, meta interface{}) error
 		return fmt.Errorf("Error checking project %q: %s", pid, err)
 	}
 
-	// Project name has changed
+	d.Partial(true)
+
+	// Project display name has changed
 	if ok := d.HasChange("name"); ok {
-		p.Name = d.Get("name").(string)
+		p.Name = project_name
 		// Do update on project
 		p, err = config.clientResourceManager.Projects.Update(p.ProjectId, p).Do()
 		if err != nil {
-			return fmt.Errorf("Error updating project %q: %s", p.Name, err)
+			return fmt.Errorf("Error updating project %q: %s", project_name, err)
 		}
+		d.SetPartial("name")
+	}
+
+	// Project parent has changed
+	if d.HasChange("org_id") || d.HasChange("folder_id") {
+		if v, ok := d.GetOk("org_id"); ok {
```

Comment:
Consider extracting line 239-253 in a `getParentResourceId` (or something along these lines) and reuse it for update and create.

Reply:
Done.